raiden_contracts/tests/test_channel_withdraw.py

```diff
@@ -0,0 +1,436 @@
+import pytest
+from eth_tester.exceptions import TransactionFailed
+from web3.exceptions import ValidationError
+from raiden_contracts.utils.events import check_withdraw
+from .fixtures.config import empty_address, fake_bytes
+from raiden_contracts.utils.config import (
+    E_CHANNEL_WITHDRAW,
+    CHANNEL_STATE_NONEXISTENT_OR_SETTLED,
+    CHANNEL_STATE_OPEN,
+    CHANNEL_STATE_CLOSED,
+    SETTLE_TIMEOUT_MIN
+)
+from .utils import MAX_UINT256
+
+
+def test_withdraw_signature(
+        token_network_test,
+        create_channel_and_deposit,
+        get_accounts,
+        create_withdraw_signatures
+):
+    (A, B) = get_accounts(2)
+    deposit_A = 5
+    deposit_B = 7
+    withdraw_A = 3
+    channel_identifier = create_channel_and_deposit(A, B, deposit_A, deposit_B)
+
+    (signature_A, signature_B) = create_withdraw_signatures(
+        [A, B],
+        channel_identifier,
+        A, withdraw_A,
+        token_network_test.address
+    )
+
+    recovered_address_A = token_network_test.call().recoverAddressFromWithdrawMessagePublic(
+        channel_identifier,
+        A, withdraw_A,
+        signature_A
+    )
+    assert recovered_address_A == A
+
+    recovered_address_B = token_network_test.call().recoverAddressFromWithdrawMessagePublic(
+        channel_identifier,
+        A, withdraw_A,
+        signature_B
+    )
+    assert recovered_address_B == B
+
+
+def test_withdraw_call(
+        token_network,
+        create_channel_and_deposit,
+        get_accounts,
+        create_withdraw_signatures
+):
+    (A, B) = get_accounts(2)
+    withdraw_A = 3
+    channel_identifier = create_channel_and_deposit(A, B, 10, 1)
+
+    (signature_A_for_A, signature_B_for_A) = create_withdraw_signatures(
+        [A, B],
+        channel_identifier,
+        A, withdraw_A
+    )
+
+    with pytest.raises(ValidationError):
+        token_network.transact({'from': A}).withdraw(
+            0x0, withdraw_A,
```

Comment:
in this and the rest of the `transact()` why are the first 2 arguments grouped together? Style guide is to keep them all in their own line. Is there a semantic reason you want them together?

Reply:
I used this way of writing only when I had a combination of short symbols / numbers (that did not took a lot of space; e.g. did not like to use an entire line for an `A`) + wanted to separate values for `A` and `B`. But I can go with the code style if needed.

Reply:
Style guide asks for the 2nd for consistency. But in general as you said we should avoid having so many arguments in functions.

I can see why you would like to group them like in the first example though.